### PR TITLE
New: AI Images - added new Tools [ED-11238]

### PR DESCRIPTION
**Pull request by bainternet**
* Remove background
* Replace background 
* Remove text

### CODE REVIEW

File: modules/ai/assets/js/editor/pages/form-media/components/generate-submit.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import { Button } from '@elementor/ui';
 
 const GenerateImagesSubmit = ( props ) => {
+	const label = props.children ?? __( 'Generate', 'elementor' );
 	return (
 		<Button fullWidth size="medium" type="submit" variant="contained" { ...props }>
-			{ __( 'Generate', 'elementor' ) }
+			{ label }
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/components/generate-submit.js:7]:
```suggestion
			{ props.children || __( 'Generate', 'elementor' ) }
```

File: modules/ai/assets/js/editor/pages/form-media/components/generate-submit.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import { Button } from '@elementor/ui';
 
 const GenerateImagesSubmit = ( props ) => {
+	const label = props.children ?? __( 'Generate', 'elementor' );
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/components/generate-submit.js:4]: You can remove this line + please change the function name to `GenerateSubmit` (my bad).

File: modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import SingleImagePreview from '../../components/single-image-preview';
+import { useEditImage } from '../../context/edit-image-context';
+import useImageActions from '../../hooks/use-image-actions';
+import useRemoveBackground from './hooks/use-remove-background';
+import NewPromptButton from '../../components/new-prompt-button';
+import { LOCATIONS } from '../../constants';
+import { useLocation } from '../../context/location-context';
+
+const RemoveBackground = () => {
+	const { editImage, width: initialEditImageWidth } = useEditImage();
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js:14]:
```suggestion
	const { editImage } = useEditImage();
```

File: modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+			</View.Panel>
+
+			<View.Content isGenerating={ isLoading }>
+				{
+					data?.result ? (
+						<div style={ {
+								backgroundImage: 'linear-gradient(45deg, #bbb 25%, transparent 25%), linear-gradient(-45deg, #bbb 25%, transparent 25%), linear-gradient(45deg, transparent 75%, #bbb 75%), linear-gradient(-45deg, transparent 75%, #bbb 75%)',
+								backgroundSize: '20px 20px',
+								backgroundPosition: '0 0, 0 10px, 10px -10px, -10px 0px',
+						} } >
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js:59]:
```suggestion
						<Box sx={ {
							backgroundImage: 'linear-gradient(45deg, #bbb 25%, transparent 25%), linear-gradient(-45deg, #bbb 25%, transparent 25%), linear-gradient(45deg, transparent 75%, #bbb 75%), linear-gradient(-45deg, transparent 75%, #bbb 75%)',
							backgroundSize: '20px 20px',
							backgroundPosition: '0 0, 0 10px, 10px -10px, -10px 0px',
						} } >
```

File: modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+							<ImagesDisplay
+								onUseImage={ use }
+								onEditImage={ edit }
+								images={ data.result }
+								aspectRatio={ editImage.aspectRatio }
+							/>
+						</div>
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js:66]:
```suggestion
						</Box>
```

File: modules/ai/assets/js/editor/pages/form-media/views/replace-background/index.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+				<ImageForm onSubmit={ handleSubmit }>
+					<PromptField
+						value={ prompt }
+						disabled={ isLoading }
+						onChange={ setPrompt }
+						placeholder={ __( 'Describe what you want to generate in the background (English only)', 'elementor' ) }
+					/>
+
+					<GenerateSubmit disabled={ isLoading } >
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/views/replace-background/index.js:49]: The button should be enabled even when there is no prompt value?

File: modules/ai/assets/js/editor/pages/form-media/components/generate-submit.js [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
-			{ __( 'Generate', 'elementor' ) }
+			{ props.children || __( 'Generate', 'elementor' ) }
 		</Button>
 	);
 };
-
-export default GenerateImagesSubmit;
+GenerateSubmit.propTypes = {
+	children: PropTypes.any,
+}
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/components/generate-submit.js:12]:
```suggestion
};

```

File: modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+			</View.Panel>
+
+			<View.Content isGenerating={ isLoading }>
+				{
+					data?.result ? (
+						<Box sx={ {
+							backgroundImage: 'linear-gradient(45deg, #bbb 25%, transparent 25%), linear-gradient(-45deg, #bbb 25%, transparent 25%), linear-gradient(45deg, transparent 75%, #bbb 75%), linear-gradient(-45deg, transparent 75%, #bbb 75%)',
+							backgroundSize: '20px 20px',
+							backgroundPosition: '0 0, 0 10px, 10px -10px, -10px 0px',
+						} } >
```
Comment (rotemee) [inline modules/ai/assets/js/editor/pages/form-media/views/remove-background/index.js:60]:
```suggestion
						} }>
```
#OCD

**Review by rotemee — DISMISSED** [review]

**Review by KingYes — DISMISSED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**